tests/unit/restapi/v1/test_groups.py

```diff
@@ -0,0 +1,373 @@
+    auth_account: dict[str, Any],
+    registered_groups: dict[str, Any],
+) -> None:
+    """Test that groups can be queried with a search term.
+
+    Given an authenticated user and registered groups, this test validates the following
+    sequence of actions:
+
+    - The user is able to retrieve a list of all registered groups with a description
+      that contains 'group'.
+    - The returned list of groups matches the expected matches from the query.
+    """
+    group1_expected = registered_groups["group1"]
+    group2_expected = registered_groups["group2"]
+    group_expected_list = [group1_expected, group2_expected]
+
+    assert_retrieving_groups_works(
+        client,
+        expected=group_expected_list,
+        search="name:*group*",
+    )
+
+
+@pytest.mark.v1
+def test_group_id_get(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_groups: dict[str, Any],
+) -> None:
+    """Test that groups can be retrieved by their unique ID.
+
+    Given an authenticated user and registered groups, this test validates the following
+    sequence of actions:
+
+    - The user is able to retrieve single group by its ID.
+    - The response is a single group with a matching ID.
+    """
+    group3_expected = registered_groups["group3"]
+
+    assert_retrieving_group_by_id_works(
+        client, group_id=group3_expected["groupId"], expected=group3_expected
+    )
+
+
+@pytest.mark.v1
+def test_cannot_register_existing_group_name(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_groups: dict[str, Any],
+) -> None:
+    """Test that registering a group with an existing name fails.
+
+    Given an authenticated user and registered groups, this test validates the following
+    sequence of actions:
+
+    - The user attempts to register a second group with the same name.
+    - The request fails with an appropriate error message and response code.
+    """
+    existing_group = registered_groups["group1"]
+
+    assert_registering_existing_group_name_fails(
+        client,
+        name=existing_group["name"],
+    )
+
+
+@pytest.mark.v1
+def test_rename_group(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_groups: dict[str, Any],
+) -> None:
+    """Test that a group can be renamed.
+
+    Given an authenticated user and registered groups, this test validates the following
+    sequence of actions:
+
+    - The user issues a request to change the name of a group.
+    - The user retrieves information about the same group and it reflects the name
+      change.
+    - The user issues a request to change the name of a group to an existing group's
+      name.
+    - The request fails with an appropriate error message and response code.
+    """
+    updated_group_name = "updated_name"
+    group_to_rename = registered_groups["group1"]
+    existing_group = registered_groups["group2"]
+
```

Comment:
nitpick: reduce whitespace

Reply:
Removed in commit: [5e972db](https://github.com/usnistgov/dioptra/pull/458/commits/5e972db226019468a4c54429528b4863cfd5366d)